lib/charms/mongodb/v0/mongos.py

```diff
@@ -158,6 +166,75 @@ def add_shard(self, shard_name, shard_hosts, shard_port=Config.MONGODB_PORT):
         logger.info("Adding shard %s", shard_name)
         self.client.admin.command("addShard", shard_url)
 
+    def remove_shard(self, shard_name: str) -> None:
+        """Removes shard from the cluster.
+
+        Raises:
+            ConfigurationError, OperationFailure, NotReadyError,
+            RemovePrimaryShardError
+        """
+        sc_status = self.client.admin.command("listShards")
+        # It is necessary to call removeShard multiple times on a shard to guarantee removal.
+        # Allow re-removal of shards that are currently draining.
+        if self._is_any_draining(sc_status, ignore_shard=shard_name):
+            cannot_remove_shard = (
+                f"cannot remove shard {shard_name} from cluster, another shard is draining"
+            )
+            logger.info(cannot_remove_shard)
+            raise NotReadyError(cannot_remove_shard)
+
+        # TODO Follow up PR, there is no MongoDB command to retrieve primary shard, this is
+        # possible with mongosh.
+        primary_shard = False
+        if primary_shard:
```

Comment:
Not sure why we have this check? 😄

Reply:
see comment above:
``` 
    # TODO Follow up PR, there is no MongoDB command to retrieve primary shard, this is
    # possible with mongosh.
```
There is a [jira ticket](https://warthogs.atlassian.net/jira/software/c/projects/DPE/boards/874/backlog?selectedIssue=DPE-2700) for this next pulse

Reply:
If you do plan to have a function that will do this check, I think it is a good idea to create a stub function for that , that will return `False` for now

Reply:
Great idea! I will do exactly that, thank you!